test/e2e/common/networking.go

```diff
@@ -28,6 +29,46 @@ var _ = ginkgo.Describe("[sig-network] Networking", func() {
 
 	ginkgo.Describe("Granular Checks: Pods", func() {
 
+		checkNodeConnectivity := func(config *e2enetwork.NetworkingTestConfig, protocol string, port int) {
+			// breadth first poll to quickly estimate failure.
+			failedPodsByHost := map[string][]*v1.Pod{}
+			// First time, we'll quickly try all pods, breadth first.
+			for _, endpointPod := range config.EndpointPods {
+				framework.Logf("Breadth first check of %v on host %v...", endpointPod.Status.PodIP, endpointPod.Status.HostIP)
+				if err := config.DialFromTestContainer(protocol, endpointPod.Status.PodIP, port, 1, 0, sets.NewString(endpointPod.Name)); err != nil {
+					if _, ok := failedPodsByHost[endpointPod.Status.HostIP]; !ok {
+						failedPodsByHost[endpointPod.Status.HostIP] = []*v1.Pod{}
+					}
+					failedPodsByHost[endpointPod.Status.HostIP] = append(failedPodsByHost[endpointPod.Status.HostIP], endpointPod)
+					framework.Logf("...failed...will try again in next pass")
+				}
+			}
+			errors := []error{}
+			// Second time, we pass through pods more carefully...
+			framework.Logf("Going to retry %v out of %v pods....", len(failedPodsByHost), len(config.EndpointPods))
+			for host, failedPods := range failedPodsByHost {
```

Comment:
if *one* node is down, all the pods from that node will be shown together.

Reply:
(imo) this isn't fundamentally changing the tests because, there is no order guarantee in the conformance definition.  that said, if we want it randomly ordered by pod, i can still do that explicitly.  

Note that the default is going to be polling like 80 times for a 6 node cluster (nodes^2 + 30) , so this test is *highly stable* in terms of retries, so the ordering isn't a sensitive thing at all.